awscrt/http.py

```diff
@@ -144,6 +145,84 @@ def on_incoming_headers_received(headers, response_code, has_body):
+                 '_on_destroy_complete', '_native_handle', 'destroy_complete')
+
+    # don't call me, I'm private
+    def __init__(self, bootstrap, on_incoming_connection, on_destroy_complete, tls_connection_options):
+        assert isinstance(bootstrap, ServerBootstrap)
+        assert tls_connection_options is None or isinstance(
+            tls_connection_options, TlsConnectionOptions)
+
+        for slot in self.__slots__:
+            setattr(self, slot, None)
+
+        self._bootstrap = bootstrap
+        self._tls_connection_options = tls_connection_options
+        self._on_incoming_connection = on_incoming_connection
+        self._on_destroy_complete = on_destroy_complete
+        self._native_handle = None
+        self.destroy_complete = Future()
+
+    
+    """
+    socket_options: awscrt.io.SocketOptions for the server's listening socket. Required
+    on_incoming_connection: Callback with signature (connection: HttpConnection.native_handle, error_code: int) Required
+    bootstrap: awscrt.io.ServerBootstrap. Required
+    tls_connection_options: awscrt.io.TlsConnectionOptions, for TLS connection
+    """
+    @staticmethod
+    def new_server(bootstrap, host_name, port, socket_options, on_incoming_connection, tls_connection_options=None):
+        """
+        Create a new server listener, binding to the host_name and port. 
+        When a new connection is received, the on_incoming_connection cb will be fired, a new ServerConnection obj will be created.
+        The aws_py_http_connection_configure_server need to be called from the callback to configure the ServerConnection 
+        """
+        assert tls_connection_options is None or isinstance(tls_connection_options, TlsConnectionOptions)
+        assert host_name is not None
+        assert port is not None
+        assert socket_options is not None and isinstance(
+            socket_options, SocketOptions)
+        assert on_incoming_connection is not None
+        if tls_connection_options is not None:
+            internal_conn_options_handle = tls_connection_options._internal_tls_conn_options
+        else:
+            internal_conn_options_handle = None
+
+        def on_destroy_complete(server_native_handle):
+            server.destroy_complete.set_result("server destroy finished!")
+        
+        server = HttpServer(bootstrap, on_incoming_connection, on_destroy_complete, tls_connection_options)
+
+        server._native_handle = _aws_crt_python.aws_py_http_server_create(
+            bootstrap._internal_bootstrap, on_incoming_connection, on_destroy_complete, host_name, port, socket_options, internal_conn_options_handle)
+        
+        return server
+
+    def close(self):
+        """
+        close the server, no more connections will be accepted, a future object will be returned, and if the close process finishes
+        the future result or exception will be set.
+        """
```

Comment:
"... **when** the close process finishes..."
Futures MUST be set to something eventually. It is a crime to never set it to anything.